http/src/response.rs

```diff
@@ -24,81 +22,90 @@ impl Response {
 		Response {
-			code: StatusCode::Forbidden,
-			content_type: header::ContentType::plaintext(),
+			code: StatusCode::FORBIDDEN,
+			content_type: HeaderValue::from_static("text/plain; charset=utf-8"),
 			content: "Provided Host header is not whitelisted.\n".to_owned(),
 		}
 	}
 
 	/// Create a response for unsupported content type.
 	pub fn unsupported_content_type() -> Self {
 		Response {
-			code: StatusCode::UnsupportedMediaType,
-			content_type: header::ContentType::plaintext(),
+			code: StatusCode::UNSUPPORTED_MEDIA_TYPE,
+			content_type: HeaderValue::from_static("text/plain; charset=utf-8"),
 			content: "Supplied content type is not allowed. Content-Type: application/json is required\n".to_owned(),
 		}
 	}
 
 	/// Create a response for disallowed method used.
 	pub fn method_not_allowed() -> Self {
 		Response {
-			code: StatusCode::MethodNotAllowed,
-			content_type: header::ContentType::plaintext(),
+			code: StatusCode::METHOD_NOT_ALLOWED,
+			content_type: HeaderValue::from_static("text/plain; charset=utf-8"),
 			content: "Used HTTP Method is not allowed. POST or OPTIONS is required\n".to_owned(),
 		}
 	}
 
 	/// CORS invalid
 	pub fn invalid_cors() -> Self {
 		Response {
-			code: StatusCode::Forbidden,
-			content_type: header::ContentType::plaintext(),
+			code: StatusCode::FORBIDDEN,
+			content_type: HeaderValue::from_static("text/plain; charset=utf-8"),
 			content: "Origin of the request is not whitelisted. CORS headers would not be sent and any side-effects were cancelled as well.\n".to_owned(),
 		}
 	}
 
 	/// Create a response for bad request
 	pub fn bad_request<S: Into<String>>(msg: S) -> Self {
 		Response {
-			code: StatusCode::BadRequest,
-			content_type: header::ContentType::plaintext(),
+			code: StatusCode::BAD_REQUEST,
+			content_type: HeaderValue::from_static("text/plain; charset=utf-8"),
 			content: msg.into()
 		}
 	}
 
 	/// Create a response for too large (413)
 	pub fn too_large<S: Into<String>>(msg: S) -> Self {
 		Response {
-			code: StatusCode::PayloadTooLarge,
-			content_type: header::ContentType::plaintext(),
+			code: StatusCode::PAYLOAD_TOO_LARGE,
+			content_type: HeaderValue::from_static("text/plain; charset=utf-8"),
 			content: msg.into()
 		}
 	}
 }
 
-impl Into<server::Response> for Response {
-	fn into(self) -> server::Response {
-		server::Response::new()
-			.with_status(self.code)
-			.with_header(self.content_type)
-			.with_body(self.content)
+// TODO: Consider removing this panicking conversion or else switch to a
+// `HttpTryFrom` or `TryFrom` conversion.
+impl From<Response> for hyper::Response<Body> {
+	/// Converts from a jsonrpc `Response` to a `hyper::Response`
+	///
+	/// ## Panics
+	///
+	/// Panics if the response cannot be converted.
+	///
+	fn from(res: Response) -> hyper::Response<Body> {
+		hyper::Response::builder()
+			.status(res.code)
+			.header("content-type", res.content_type)
+			.body(res.content.into())
+			.expect("Unable to convert response")
```

Comment:
I don't think it will ever fail. I would leave `expect` with a more detailed proof, why we believe that response creation can never fail (i.e. all the data provided in other fields is valid).

Reply:
I'm not sure I'd go so far as to say it could never fail, particularly if changes are made to `Response` so I'm not sure if a detailed proof would be appropriate. It might be best just to wait for [`TryFrom`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html) to stabilize. Let me know what you think.